src/devices/machine/pit8253.cpp

```diff
@@ -1137,3 +1172,49 @@ void pit_counter_device::set_clock_signal(int state)
 	}
 	m_clock_signal = state;
 }
+
+uint8_t ps2_pit_device::read(offs_t offset)
+{
+	LOG2("read(): offset %d\n", offset);
+
+	if (offset == 3)
+	{
+		/* Reading mode control register is illegal according to docs */
+		/* Experimentally determined: reading it returns 0 */
+		return 0;
+	}
+	else if(offset == 4) // register 4 is timer channel 3.
+	{
+		return m_ch3_counter->read();
+	}
+	else if(offset <= 2)
+		return m_counter[offset]->read();
+	else
+		return 0xFF;
```

Comment:
You’re not being consistent with `if (...)` vs `if(...)` in this function.  Also, MAME convention is lowercase hex digits.